### PR TITLE
Allow run_tactic to return a value, remove hack from ltac2

### PR DESCRIPTION
**Pull request by SkySkimmer**
Overlay: https://github.com/coq-community/paramcoq/pull/29

### CODE REVIEW

**Review by ppedrot — APPROVED** [review]
Code is fine. It might even be reasonable for `run_tactic` to return a value, so as not to export another variant. WDYT?

**Comment by SkySkimmer** [comment]
>Code is fine. It might even be reasonable for run_tactic to return a value, so as not to export another variant. WDYT?

We can go further, see 2nd commit. Not sure if want.

**Comment by ppedrot** [comment]
The second commit is slightly exaggerated, I don't think you want those things to return values in general.

File: proofs/proof.mli [hunk 1]
```diff
@@ -219,7 +219,7 @@ val no_focused_goal : t -> bool
    used. In which case it is [false]. *)
 val run_tactic
   :  Environ.env
-  -> unit Proofview.tactic -> t -> t * (bool*Proofview_monad.Info.tree)
+  -> 'a Proofview.tactic -> t -> t * (bool*Proofview_monad.Info.tree) * 'a
```
Comment (ppedrot) [inline proofs/proof.mli]: Can you revert the other API changes and only leave that one? I doubt that the other functions should return values.

Comment (SkySkimmer) [reply proofs/proof.mli]: Changed

**Review by ppedrot — APPROVED** [review]

**Pull request merged**